BUILD.bazel

```diff
@@ -0,0 +1,86 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "srcs",
+    srcs = glob(
+        [
+            "src/snmalloc/**/*",
+            "src/test/*.h",
+            "CMakeLists.txt",
+        ],
+    ),
+    visibility = ["//visibility:private"],
+)
+
+cmake(
+    name = "snmalloc",
+    cache_entries = {
+        "CMAKE_BUILD_TYPE": "Release",
```

Comment:
Question: to me it is hardcoded or is it possible to get the value of `--compilation_mode` for example ?

Reply:
Not really a question but ideally (does not have to be in this PR though ?) would expect something like

```
...
"CMAKE_BUILD_TYPE": select({
        ":compilation_mode_fastbuild": "RelWithDebInfo",  # or "None" if you want no specific type
        ":compilation_mode_opt": "Release",
        ":compilation_mode_dbg": "Debug",
        "//conditions:default": "Release",
    }),
...
```